addon/serializers/json-serializer.js

```diff
@@ -716,10 +716,11 @@ export default Serializer.extend({
   */
   normalizeUsingDeclaredMapping: function(modelClass, hash) {
     var attrs = get(this, 'attrs');
-    var payloadKey, normalizedKey, key;
+    var payloadKey, key;
 
     if (attrs) {
       for (key in attrs) {
+        var normalizedKey = null;
```

Comment:
I think we could leave the declaration in top and just set `normalizedKey = payloadKey` in the `for`. That way we don't have to do the extra check on line 736.